### PR TITLE
Setup project

### PR DESCRIPTION
**Pull request by fellainthewagon**
- create dynamically config: app will use one of two types of storage (memory(RAM) and Redis).  See README how to.
- create factory
- implements DI (i hope it is lol)
- added volumes, env vars and description to README.md

### CODE REVIEW

File: project/client/public/style.css [hunk 1]
```diff
@@ -0,0 +1,75 @@
+* {
```
Comment (iantiporuk) [inline project/client/public/style.css:1]: можешь посмотреть в сторону bootstrap, просто совет

File: project/server/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+node_modules
+npm-debug.log
+Dockerfile
+__tests__
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply project/server/.dockerignore:4]: аа, хорошо, закину, и все исправлю что было выше

File: project/server/src/schema/ajv.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const Ajv = require("ajv");
+const addFormats = require("ajv-formats");
+
+const ajv = new Ajv({ allErrors: true });
+
+// все делаю по доке, но вылетает ошибка при добалвении форматов
```
Comment (iantiporuk) [inline project/server/src/schema/ajv.js:6]: russian is prohibited

File: project/server/src/app.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+const errorHandler = require("./middleware/errorHandler");
+const cors = require("./middleware/cors");
+
+const app = express();
+
+/**
+ * middleware
+ */
+app.use("/api-docs", swaggerUI.serve, swaggerUI.setup(swaggerDocs));
+app.use(morgan("dev"));
```
Comment (iantiporuk) [inline project/server/src/app.js:18]: я думаю, ты знаешь что надо сделать с dev

File: project/server/src/components/resolutions/ResolutionsController.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  async getResolution(req, res, next) {
+    try {
+      const resolution = await resolutionsService.getResolution(req.params.key);
+      if (resolution) return res.json(resolution.data);
+      return res.status(404).json({ message: "Resolution not found" });
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply project/server/src/components/resolutions/ResolutionsController.js]: fixed

File: project/server/src/components/resolutions/ResolutionsController.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+  async deleteResolution(req, res, next) {
+    try {
+      const isDeleted = await resolutionsService.deleteResolution(
+        req.params.key
+      );
+      if (isDeleted) {
+        return res.json({ message: "Resolution successfully deleted" });
+      }
+      return res.status(404).json({ message: "Resolution not found" });
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply project/server/src/components/resolutions/ResolutionsController.js]: fixed

File: project/server/.dockerignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply project/server/.dockerignore]: fixed

File: project/client/public/style.css (hunk 1 above)
Comment (fellainthewagon) [reply project/client/public/style.css]: ok, later :)

File: project/client/public/Resolution.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  async delete(key) {
+    try {
+      const response = await fetch(this.url + key + "/resolution", {
+        method: "DELETE",
+      });
+      return response.json();
+    } catch (error) {
+      console.log(error);
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply project/client/public/Resolution.js]: fixed

File: project/client/public/Resolution.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    }
+  }
+
+  async find(key) {
+    try {
+      const response = await fetch(this.url + key + "/resolution");
+      return response.json();
+    } catch (error) {
+      console.log(error);
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply project/client/public/Resolution.js]: fixed

File: project/client/public/Resolution.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+      const response = await fetch(this.url + key + "/resolution", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(value),
+      });
+      return response.json();
+    } catch (error) {
+      console.log(error);
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply project/client/public/Resolution.js]: fixed

File: project/client/public/Handlers.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    e.preventDefault();
+    const { key } = await Queue.getCurrent();
+    if (!key) return;
+    await Resolution.add(key, { resolution: resolutionInput.value });
+    resolutionInput.value = "";
+  }
+
+  async findResolution(e) {
+    e.preventDefault();
+    name = findInputQueue.value
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply project/client/public/Handlers.js]: fixed

File: project/client/public/Handlers.js [hunk 11]
```diff
@@ -0,0 +1,72 @@
+import Resolution from "./Resolution.js";
+import Queue from "./Queue.js";
+
+const showCurrentPatient = document.querySelectorAll(".current-patient");
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply project/client/public/Handlers.js]: fixed

File: project/client/package.json [hunk 12]
```diff
@@ -0,0 +1,15 @@
+{
+  "scripts": {
+    "start": "http-server -a localhost -p 8000 -c-1"
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply project/client/package.json]: fixed

File: README.md [hunk 13]
```diff
@@ -1,13 +1,44 @@
-# Project "Doctors cabinete"
+# Project "Clinic"
 
-A sigle Express-based app with two unrelated functionality:
+run:
 
-1. In-memory stack (LIFO)
+```sh
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply README.md:19]: done

File: README.md [hunk 14]
```diff
@@ -1,13 +1,44 @@
-# Project "Doctors cabinete"
+# Project "Clinic"
 
-A sigle Express-based app with two unrelated functionality:
+run:
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply README.md]: done

File: project/server/src/components/resolutions/ResolutionsService.js [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+  constructor() {
+    this.storage = [
+      /*       {
+        data: { key: "vincent", resolution: "drag addict" },
+        timestamp: 1728346053674,
+      }, */
+    ];
+  }
+
+  async addResolution(value, ttl = 30000) {
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply project/server/src/components/resolutions/ResolutionsService.js]: fixed

File: project/server/src/middleware/cors.js [hunk 16]
```diff
@@ -0,0 +1,9 @@
+module.exports = (req, res, next) => {
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply project/server/src/middleware/cors.js]: fixed

File: project/server/src/schema/ajv.js (hunk 3 above)
Comment (fellainthewagon) [reply project/server/src/schema/ajv.js]: totally agree

File: project/server/src/app.js (hunk 4 above)
Comment (fellainthewagon) [reply project/server/src/app.js]: done

File: server/__tests__/queue.spec.js [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+    expect(res.status).toBe(200);
+  });
+
+  it("returns body.message when queue is empty", async () => {
+    const res = await getPersonFromQueue();
+    expect(res.body.message).toBe("The Queue is empty");
+  });
+
+  /**
+   * здесь метод "get" я реализовал таким образом, что он забирает
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply server/__tests__/queue.spec.js]: i will delet all this tests

File: docker-compose.yml [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+    volumes:
+      - ./server:/app
+    ports:
+      - "3000:3000"
+    networks:
+      - clinic-app
+    depends_on:
+      - redis
+    environment:
+      DEV_REDIS_HOST: redis
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply docker-compose.yml:20]: dont know how up app without this var
i tried many solutions which found in internet, but only that works...

[the remark this answers is not in the document]
Comment (fellainthewagon) [reply docker-compose.yml:20]: learn about docker-compose container_name and hostname - ok, I’ll see if I can write the tests today

File: server/.env.example [hunk 19]
```diff
@@ -0,0 +1,8 @@
+NODE_ENV=dev
+DEV_PORT=3000
+DEV_HOST=0.0.0.0
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply server/.env.example:3]: I've seen it somewhere, ok i fix it

File: server/src/library/statusMessage.js [hunk 20]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply server/src/library/statusMessage.js:1]: `lib`?)

File: server/src/components/storageFactory/index.js [hunk 21]
```diff
@@ -0,0 +1,23 @@
+const container = require("./storageContainer");
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply server/src/components/storageFactory/index.js:1]: refactored

**Comment by fellainthewagon** [comment]
ы

**Comment by fellainthewagon** [comment]
I removed unnecess modules and made a common class for each type of storage, injected factory into services

**Pull request merged**